src/com/qamp/HarisJasarevic/homeworks/homework5/task1/ATM.java

```diff
@@ -0,0 +1,55 @@
+package com.qamp.HarisJasarevic.homeworks.homework5.task1;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ATM {
+
+    private List<BankAccounts> accounts;
+
+    public ATM () {
+        this.accounts = new ArrayList<>();
+    }
+
+    public double moneyWithdraw (final double withdrawAmount, final String serialNumber, final int passCode) {
+        final BankAccounts account = getAccount(serialNumber, passCode);
+        return account.moneyWithdraw(withdrawAmount);
+    }
+
+    public double moneyDeposit ( final double depositAmount, final String serialNumber, final int passCode) {
+        final BankAccounts account = getAccount (serialNumber, passCode);
+        return account.moneyDeposit(depositAmount);
+    }
+
+    public void reset (final String serialNumber, final int passCode) throws IllegalStateException {
+        final BankAccounts account = getAccount (serialNumber, passCode);
+        account.reset();
+    }
+
+    public String showAccountState (final String serialNumber, final int passCode) throws IllegalStateException {
+        final BankAccounts account = getAccount (serialNumber, passCode);
+        return account.showAccountState();
+    }
+
+    private BankAccounts getAccount(final String serialNumber, final int passCode) throws IllegalStateException {
+        for (BankAccounts account : accounts) {
+            if (account.serialNumber.equals(serialNumber)) {
+                if (passCode >= 1000 && passCode <= 9999) {
```

Comment:
this check should throw an IllegalArgumentException instead of IllegalState as per task description